subschemas/substrate-api-proxy/src/resolvers/Query/treasury.ts

```diff
@@ -28,19 +30,41 @@ export async function treasurySummary(
 
   return {
     activeProposals: proposals.proposals.length,
-    proposalCount: proposals.proposalCount.toString(),
+    totalProposals: proposals.proposalCount.toNumber(),
     approvedProposals: proposals.approvals.length,
-    spendPeriod: api.consts.treasury.spendPeriod.toString(),
+    spendPeriod: createSpendPeriod(api, bestNumber),
     treasuryBalance: {
       accountId: treasuryBalance.accountId.toString(),
       accountNonce: treasuryBalance.accountNonce.toString(),
-      freeBalance: treasuryBalance.freeBalance.toString(),
-      frozenFee: treasuryBalance.frozenFee.toString(),
-      frozenMisc: treasuryBalance.frozenMisc.toString(),
-      reservedBalance: treasuryBalance.reservedBalance.toString(),
-      votingBalance: treasuryBalance.votingBalance.toString(),
+      freeBalance: formatBalance(api, treasuryBalance.freeBalance),
```

Comment:
I did something similar with the format balance but instead passing the registry. 
So doing `formatBalance` X amount of times, aren't you query the API those amount of times? 

Wouldn't it be better to get the registry **once** and pass the registry to the function? 
Like so:
https://github.com/litentry/litentry-graph/pull/109/files#diff-efaf21b3700988dcc51fc2f45e46b102278079fede4a0922dfcd718ce1cf366cR77

Do you have any thoughts on that? :)

Reply:
Maybe `api.registry` is not an async call and I've overengineered 🤔

Reply:
`api.registry` does not make any network call, it's a constant created when the api is initialized